modules/@apostrophecms/doc-type/index.js

```diff
@@ -789,6 +771,72 @@ module.exports = {
         });
         return draft;
       },
+
+      // Unpublish a document as well as its previous version if any,
+      // and update the draft version.
+      // This method accepts the draft or the published version of the document
+      // to achieve this.
+      async unpublish(req, doc) {
+        const DRAFT_SUFFIX = ':draft';
+        const PUBLISHED_SUFFIX = ':published';
+        const PREVIOUS_SUFFIX = ':previous';
+
+        const isDocDraft = doc._id.includes(DRAFT_SUFFIX);
+        const isDocPublished = doc._id.includes(PUBLISHED_SUFFIX);
+
+        if (!isDocDraft && !isDocPublished) {
+          // TODO: throw?
+          return;
+        }
+
+        const published = isDocPublished
+          ? doc
+          : await self.apos.doc.db.findOne({
+            _id: doc._id.replace(DRAFT_SUFFIX, PUBLISHED_SUFFIX)
+          });
+
+        const draft = isDocDraft
+          ? doc
+          : await self.apos.doc.db.findOne({
+            _id: doc._id.replace(PUBLISHED_SUFFIX, DRAFT_SUFFIX)
+          });
+
+        if (!draft) {
+          // TODO: throw?
+          return;
+        }
+
+        const previous = await self.apos.doc.db.findOne({
+          _id: published._id.replace(PUBLISHED_SUFFIX, PREVIOUS_SUFFIX)
+        });
+
+        // TODO: `manager.emit`?
+        self.emit('beforeUnpublish', req, published);
+
+        await self.apos.doc.db.updateOne(
+          { _id: draft._id },
+          {
+            $set: {
+              // TODO:: set `modified: 1`?
+              modified: 1,
+              // TODO: unset `lastPublishedAt`?
```

Comment:
in https://github.com/apostrophecms/apostrophe/pull/3754/files#diff-5e6c997e317e3d9d022272dd5b800a9ddd9965d767280831bd1148f202c1dc1dL426-L428, the `lastPublishedAt` was unset.

Would it be better to unset it rather to set it to `null`?